src/panels/lovelace/components/hui-yaml-editor.ts

```diff
@@ -26,25 +30,52 @@ export class HuiYamlEditor extends HTMLElement {
     const shadowRoot = this.attachShadow({ mode: "open" });
     shadowRoot.innerHTML = `
             <style>
-               ${codeMirrorCSS}
-               .CodeMirror {
-                  height: var(--code-mirror-height, 300px);
-                  direction: var(--code-mirror-direction, ltr);
-                }
-                .CodeMirror-gutters {
-                  border-right: 1px solid var(--paper-input-container-color, var(--secondary-text-color));
-                  background-color: var(--paper-dialog-background-color, var(--primary-background-color));
-                  transition: 0.2s ease border-right;
-                }
-                .CodeMirror-focused .CodeMirror-gutters {
-                  border-right: 2px solid var(--paper-input-container-focus-color, var(--primary-color));;
-                }
-                .CodeMirror-linenumber {
-                  color: var(--paper-dialog-color, var(--primary-text-color));
-                }
+              ${codeMirrorCSS}
+              .CodeMirror {
+                height: var(--code-mirror-height, 300px);
+                direction: var(--code-mirror-direction, ltr);
+              }
+              .CodeMirror-gutters {
+                border-right: 1px solid var(--paper-input-container-color, var(--secondary-text-color));
+                background-color: var(--paper-dialog-background-color, var(--primary-background-color));
+                transition: 0.2s ease border-right;
+              }
+              .CodeMirror-focused .CodeMirror-gutters {
+                border-right: 2px solid var(--paper-input-container-focus-color, var(--primary-color));;
+              }
+              .CodeMirror-linenumber {
+                color: var(--paper-dialog-color, var(--primary-text-color));
+              }
             </style>`;
   }
 
+  set hass(hass: HomeAssistant) {
+    if (
+      (!this._hass || this._hass.language !== hass.language) &&
+      this.shadowRoot
+    ) {
+      if (computeRTL(hass)) {
+        this.shadowRoot.innerHTML += `
+            <style id="rtlStyle">
+              .CodeMirror-vscrollbar {
```

Comment:
Let's not append/remove CSS nodes. Let's do it like with the other RTL elements, and just change a class using `this.classSet` + use scoped CSS. `.rtl .CodeMirror-vscrollbar { }`